allianceutils/api/permissions.py

```diff
@@ -65,6 +65,13 @@ class GenericDjangoViewsetPermissions(BasePermission):
     def __init__(self):
         self._saved_actions_to_perms_map = None
 
+    def get_model(self, view):
+        """
+        Get the model used for permission check. By default tries to derive model from queryset
+        Can be overriden in case a different model is needed or get_queryset is not available
+        """
+        return view.get_queryset().model
```

Comment:
I think we discussed on slack:

leave `get_model` on permission but maybe:

```
def get_model(self, view):
    if hasattr(view, 'get_permission_model'):
        return view.get_permission_model()
    return view.get_queryset().model
```

that way can either override `get_model` on permission class or add `get_permission_model` to viewset